plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -201,6 +201,56 @@ public Stream<DeltaLakeTransactionLogEntry> getCheckpointTransactionLogEntries(
                     checkpoint));
         }
 
+        return getCheckpointTransactionLogEntries(
+                session,
+                entryTypes,
+                checkpointSchemaManager,
+                typeManager,
+                fileSystem,
+                stats,
+                checkpoint,
+                metadataAndProtocol.map(MetadataAndProtocolEntry::metadataEntry),
+                metadataAndProtocol.map(MetadataAndProtocolEntry::protocolEntry));
+    }
+
+    public Stream<DeltaLakeTransactionLogEntry> getCheckpointTransactionAddFileEntries(
```

Comment:
I don't think we need overload `getCheckpointTransactionLogEntries`.

The existing method just needs `Optional<MetadataAndProtocolEntry>` parameter.

It's called in just 2 places, so no such burden, especially compared with ~4 overloads (public + private), with long cascades of similar (but not equal) param lists